### ISSUE TITLE
jcmd help output is not sufficient

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#6780]
```
Format:
    jcmd <vmid> <arguments>
 Arguments:
    <vmid>: Attach API VM ID as shown in jps or other Attach API-based tools.
    command[command arguments]: command from the list returned by "-help"
    -help: print the list of diagnostic commands
```

- It's not obvious that `jcmd <vmid> help` prints the available commands. 
- How do you get help on the individual commands? i.e. `help Dump.heap` works
- When you run `jcmd <vmid>` without any command it results in an error. It may be useful to have this print the available commands.

**Comment by pshipton** [comment]
@pdbain-ibm @JasonFengJ9

**Comment by JasonFengJ9** [comment]
Will have a look.

**Comment by JasonFengJ9** [comment]
> It's not obvious that jcmd <vmid> help prints the available commands.

`jcmd help` has following output:
```
jcmd: run diagnostic command on another Java process
 Format:
    jcmd <vmid> <arguments>
 Arguments:
    <vmid>: Attach API VM ID as shown in jps or other Attach API-based tools.
    command[command arguments]: command from the list returned by "-help"
    -help: print the list of diagnostic commands .
    -J: supply arguments to the Java VM running jcmd
```

`    -help: print the list of diagnostic commands .` could be changed to `    -help: print the list of available diagnostic commands .`, does this make it clearer or any suggestions?

`RI jcmd help` gives following output:
```
Could not find any processes matching : 'help'
```

`jcmd 33758 help` gives:
```
33758:
The following commands are available:
...
GC.class_histogram
GC.run
...
help
```

**Comment by pshipton** [comment]
We need to document that `jcmd <vmid> help` prints the list of available commands, and that `jcmd <vmid> help <command>` prints help on the individual command.

**Comment by pshipton** [comment]
I don't have any issue with the `jcmd -help` command and what it already prints, but we need to include more information.

**Comment by pshipton** [comment]
I guess part of the problem is that `jcmd <vmid> -help` gives an error as the proper format is `jcmd <vmid> help`

**Comment by pshipton** [comment]
The detail under `Arguments:` is confusing, is this supposed to mean the values for `<arguments>`? Some are and some are not. `-J` is not a valid `<arguments>` and neither is `<vmid>` .

**Comment by JasonFengJ9** [comment]
> Part of the problem is that jcmd <vmid> -help gives an error as the proper format is jcmd <vmid> help

Are you suggesting that `-help` should be allowed or the error message need a better description than `Error parsing properties: java.io.IOException: key openj9_diagnostics.errortype not found`?

**Comment by pshipton** [comment]
I don't expect `-help` to be supported, unless OpenJDK supports this format as well.

**Comment by JasonFengJ9** [comment]
> The detail under Arguments: is confusing, is this supposed to mean the values for <arguments>? Some are and some are not. -J is not a valid <arguments> and neither is <vmid> .

How about change the `help` output to something like following:
```
Usage : jcmd <generalOption> <vmid> <arguments>
 generalOption:
    -J : supply arguments to the Java VM running jcmd
    -l :  list JVM processes on the local machine
    -h --help : print this help message

<vmid> : Attach API VM ID as shown in jps or other Attach API-based tools.

arguments:
    help : print the list of diagnostic commands
    command [command arguments] : command from the list returned by "help"
    
If no options are given, Java processes are listed (same as -l).

 NOTE: this utility may significantly affect the performance of the target JVM.
     The available diagnostic commands are determined by
     the target VM and may vary between VMs.
```

**Comment by pshipton** [comment]
It's better. Not sure we need the `generalOption:` header since the other utilities don't use this. Similarly for `-h --help`, the other utilities don't list these. I think any option which isn't understood results in help.

The part which is still missing is `help <command>`, which prints help for the specific command.

**Comment by JasonFengJ9** [comment]
> Not sure we need the generalOption: header since the other utilities don't use this. 

`RI jstat` uses it [1]. Other OpenJ9 utilities could add it later. 

> Similarly for -h --help, the other utilities don't list these. 

`RI jcmd` has these options. Other utilities could add it as well.

> I think any option which isn't understood results in help.

Yes, this is an existing behaviour.

[1] https://docs.oracle.com/javase/7/docs/technotes/tools/share/jstat.html

Added `help <command>`.
```
Usage : jcmd <generalOption> <vmid> <arguments>
 generalOption:
    -J : supply arguments to the Java VM running jcmd
    -l :  list JVM processes on the local machine
    -h --help : print this help message

<vmid> : Attach API VM ID as shown in jps or other Attach API-based tools.

arguments:
    help : print the list of diagnostic commands
    help command : print help for the specific command
    command [command arguments] : command from the list returned by "help"
    
If no options are given, Java processes are listed (same as -l).

 NOTE: this utility may significantly affect the performance of the target JVM.
     The available diagnostic commands are determined by
     the target VM and may vary between VMs.
```

**Comment by pshipton** [comment]
Please remove `generalOption:` and just list them before `arguments:`. Otherwise looks good. I don't think the `-h --help` is needed, but if you really want to add it I guess it doesn't hurt. Technically help is printed when any invalid option is used. I'd only use a single dash for `-help`.

**Comment by pshipton** [comment]
Or leave out `-help` and just use `-h`

**Comment by JasonFengJ9** [comment]
Modified as per comments:

```
Usage : jcmd <vmid> <arguments>
    -J : supply arguments to the Java VM running jcmd
    -l :  list JVM processes on the local machine
    -h : print this help message

<vmid> : Attach API VM ID as shown in jps or other Attach API-based tools.

arguments:
    help : print the list of diagnostic commands
    help command : print help for the specific command
    command [command arguments] : command from the list returned by "help"
    
If no options are given, Java processes are listed (same as -l).

 NOTE: this utility may significantly affect the performance of the target JVM.
     The available diagnostic commands are determined by
     the target VM and may vary between VMs.
```

**Comment by pshipton** [comment]
I think the `<vmid>` line should be indented to line up with the -h option, otherwise looks good.

**Comment by pshipton** [comment]
The following ends with a period `Attach API VM ID as shown in jps or other Attach API-based tools.` but no other option description does.

**Comment by pshipton** [comment]
There is an extra space after the colon in `l :  list JVM processes on the local machine`

**Comment by pshipton** [comment]
` list JVM processes on the local machine. Default behavior when no options are specified.`

**Comment by JasonFengJ9** [comment]
Thanks @pshipton, will submit a PR and fix format there.

**Comment by pshipton** [comment]
`help command` should be `help <command>`

**Comment by pshipton** [comment]
and `command [command arguments]` should be `<command> [command arguments]`

**Comment by JasonFengJ9** [comment]
Submit https://github.com/eclipse/openj9/pull/6789 for format changing.

> When you run jcmd <vmid> without any command it results in an error. It may be useful to have this print the available commands.

Will address this in another PR.

**Comment by pshipton** [comment]
Fixed by #6789

**Comment by JasonFengJ9** [comment]
> When you run jcmd <vmid> without any command it results in an error. It may be useful to have this print the available commands.

Created PR https://github.com/eclipse/openj9/pull/6804
